bundles/org.openhab.binding.sbus/src/main/java/org/openhab/binding/sbus/handler/SbusSwitchHandler.java

```diff
@@ -0,0 +1,173 @@
+            return;
+        }
+
+        try {
+            SbusDeviceConfig config = getConfigAs(SbusDeviceConfig.class);
+            int[] statuses = adapter.readStatusChannels(config.subnetId, config.id);
+            if (statuses == null) {
+                logger.warn("Received null status channels from SBUS device");
+                return;
+            }
+
+            // Iterate over all channels and update their states
+            for (Channel channel : getThing().getChannels()) {
+                SbusChannelConfig channelConfig = channel.getConfiguration().as(SbusChannelConfig.class);
+                if (channelConfig.channelNumber > 0 && channelConfig.channelNumber <= statuses.length) {
+                    String channelTypeId = channel.getChannelTypeUID().getId();
+                    boolean isActive = statuses[channelConfig.channelNumber - 1] == 0x64; // 100 when on, something else
+                                                                                          // when off
+
+                    if ("switch-channel".equals(channelTypeId)) {
+                        updateState(channel.getUID(), isActive ? OnOffType.ON : OnOffType.OFF);
+                    } else if ("dimmer-channel".equals(channelTypeId)) {
+                        updateState(channel.getUID(), new PercentType(statuses[channelConfig.channelNumber - 1]));
+                    } else if ("paired-channel".equals(channelTypeId)) {
+                        updateState(channel.getUID(), isActive ? OpenClosedType.OPEN : OpenClosedType.CLOSED);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Error reading status channels", e);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        final SbusAdapter adapter = super.sbusAdapter;
+        if (adapter == null) {
+            logger.warn("SBUS adapter not initialized");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "SBUS adapter not initialized");
+            return;
+        }
+
+        try {
+            Channel channel = getThing().getChannel(channelUID);
+            if (channel != null) {
+                SbusChannelConfig channelConfig = channel.getConfiguration().as(SbusChannelConfig.class);
+                if (channelConfig.channelNumber <= 0) {
+                    logger.warn("Invalid channel number for {}", channelUID);
+                    return;
+                }
+
+                SbusDeviceConfig config = getConfigAs(SbusDeviceConfig.class);
+
+                if (command instanceof OnOffType) {
+                    handleOnOffCommand((OnOffType) command, config, channelConfig, channelUID, adapter);
+                } else if (command instanceof PercentType) {
+                    handlePercentCommand((PercentType) command, config, channelConfig, channelUID, adapter);
```

Comment:
Prevent double casting, also applies to the others in this if/else block (use java 17 instanceof pattern matching)
```suggestion
                } else if (command instanceof PercentType commandAsPercentType) {
                    handlePercentCommand(commandAsPercentType, config, channelConfig, channelUID, adapter);
```